packages/next-app/src/app/metadata/[poolId]/page.tsx

```diff
@@ -1,24 +1,82 @@
 "use client";
+
+import { useRef, useState } from "react";
+
+import { Button } from "../../../components";
 import metadataGql from "../../../lib/poolMetadataGql";
 
 export default function Page({ params }: { params: { poolId: string } }) {
+  const textField = useRef<HTMLTextAreaElement>(null);
+
+  const [metadata, setMetadata] = useState(`[
+    {
+      "name": "Pool Address",
+      "type": "address",
+      "desc": "The address of the smart contract that implements the exchange pool",
+      "value": "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D"
+    },
+    {
+      "name": "Pool link",
+      "type": "URL",
+      "desc": "The address of the smart contract that implements the exchange pool",
+      "value": "https://github.com/"
+    }
+  ]`);
```

Comment:
Couldn't we just JSON.dump this before pinning instead of adding this to a string?

Reply:
It was getting annoying be swapping between JSON.stringify and JSON.parse on fetch/route handler for this test. Because I got some obscured errors. I didn't spend time on this and let it on string format. 🚀